vyper/semantics/validation/local.py

```diff
@@ -490,53 +496,67 @@ def visit_Log(self, node):
 
 
 class _LocalExpressionVisitor(VyperNodeVisitorBase):
-    ignored_types = (vy_ast.Constant, vy_ast.Name)
```

Comment:
doesn't `vy_ast.Constant` encompass all of the new ignored types (Str, Bytes, Decimal, etc)?

Reply:
It is over-inclusive because it will catch `vy_ast.Int` nodes as well, which we want to visit in this case.

Reply:
honestly the special logic for visiting `Int` nodes looks a little sus? it does annotation during what should just be a validation pass.

Reply:
hm yes that is true. I will check again if this can be performed in annotation.